src/main/java/co/rsk/federate/signing/hsm/client/HSMClientProtocolFactory.java

```diff
@@ -0,0 +1,36 @@
+package co.rsk.federate.signing.hsm.client;
+
+import co.rsk.federate.config.SignerConfig;
+import co.rsk.federate.rpc.SocketBasedJsonRpcClientProvider;
+import co.rsk.federate.signing.hsm.HSMUnsupportedTypeException;
+
+import java.net.InetSocketAddress;
+
+public class HSMClientProtocolFactory {
+    public static final String HOST = "host";
+    public static final String PORT = "port";
+    public static final String SOCKET_TIMEOUT = "socketTimeout";
+    public static final String MAX_ATTEMPTS = "maxAttempts";
+    public static final String INTERVAL_BETWEEN_ATTEMPTS = "intervalBetweenAttempts";
+
+    public static final int DEFAULT_SOCKET_TIMEOUT = 10_000;
+    public static final int DEFAULT_ATTEMPTS = 2;
+    public static final int DEFAULT_INTERVAL = 1000;
+
+    public HSMClientProtocol buildHSMClientProtocolFromConfig(SignerConfig config) throws HSMUnsupportedTypeException {
+        if (!"hsm".equalsIgnoreCase(config.getType())) {
+            throw new HSMUnsupportedTypeException("Config type must be HSM");
```

Comment:
If there are different types of config, why not create a `HsmConfig` class that extends from SignerConfig so this constructor receives only that type of config?